js_modules/dagit/packages/core/src/runs/usePipelineRunLogsSubscription.tsx

```diff
@@ -0,0 +1,59 @@
+import {useSubscription} from '@apollo/client';
+import React from 'react';
+
+import {AppContext} from '../app/AppContext';
+import {arrayBufferToString} from '../workers/util';
+
+import {PIPELINE_RUN_LOGS_SUBSCRIPTION} from './PipelineRunLogsSubscription';
+import {
+  PipelineRunLogsSubscription,
+  PipelineRunLogsSubscriptionVariables,
+} from './types/PipelineRunLogsSubscription';
+
+export function usePipelineRunLogsSubscription({runId, cursor}: any, onLogs: any) {
+  useSubscription<PipelineRunLogsSubscription, PipelineRunLogsSubscriptionVariables>(
+    PIPELINE_RUN_LOGS_SUBSCRIPTION,
+    {
+      fetchPolicy: 'no-cache',
+      variables: {runId, cursor},
+      onSubscriptionData: ({subscriptionData}) => {
+        const logs = subscriptionData.data?.pipelineRunLogs;
+        if (!logs || logs.__typename === 'PipelineRunLogsSubscriptionFailure') {
+          return;
+        }
+        onLogs(logs);
+      },
+    },
+  );
+}
+
+export function usePipelineRunLogsSubscriptionWorker({runId}: any, onLogs: any) {
```

Comment:
Would be really nice to have the types tightened up on these.

Reply:
will do